mempool/src/pool/mod.rs

```diff
@@ -636,7 +636,7 @@ where
     pub async fn add_transaction(&mut self, tx: SignedTransaction) -> Result<(), Error> {
         let conflicts = self.validate_transaction(&tx).await?;
         self.store.drop_conflicts(conflicts);
-        self.finalize_tx(tx).await?;
+        self.finalize_tx(tx.clone()).await?;
```

Comment:
I don't think this is necessary

Reply:
Sure, it is an artifact from the previous implementation attempt. I have even removed it locally, but forgot to push the final commit before marking the pull request as ready. Sorry, fixed.